aries_cloudagent/protocols/coordinate_mediation/mediation_invite_store.py

```diff
@@ -0,0 +1,165 @@
+        except StorageNotFoundError:
+            return None
+
+    async def store(
+        self, mediation_invite: MediationInviteRecord
+    ) -> MediationInviteRecord:
+        """
+        Store the mediator's invite for further use when starting the agent.
+
+        Update the currently stored invite if one already exists.
+        This assumes a new invite and as such, marks it as unused.
+
+        :param mediation_invite: mediation invite url
+        :return: stored mediation invite
+        """
+        current_invite_record = await self.__retrieve_record(self.MEDIATION_INVITE_ID)
+
+        if current_invite_record is None:
+            await self.__storage.add_record(
+                StorageRecord(
+                    type=self.INVITE_RECORD_CATEGORY,
+                    id=self.MEDIATION_INVITE_ID,
+                    value=mediation_invite.to_json(),
+                )
+            )
+        else:
+            await self.__storage.update_record(
+                current_invite_record,
+                mediation_invite.to_json(),
+                tags=current_invite_record.tags,
+            )
+
+        return mediation_invite
+
+    async def __retrieve(self) -> Optional[MediationInviteRecord]:
+        """:return: the currently stored mediation invite url."""
+
+        invite_record = await self.__retrieve_record(self.MEDIATION_INVITE_ID)
+        return (
+            MediationInviteRecord.from_json(invite_record.value)
+            if invite_record is not None
+            else None
+        )
+
+    async def __update_mediation_record(
+        self, provided_mediation_invitation: str
+    ) -> MediationInviteRecord:
+        """
+        Update the stored invitation when a new invitation is provided.
+
+        Stored value is only updated if `provided_mediation_invitation` has changed.
+        Updated record is marked as unused.
+
+        :param provided_mediation_invitation: mediation invite provided by user
+        :return: stored mediation invite
+        """
+        default_invite = await self.__retrieve()
+
+        if default_invite != provided_mediation_invitation:
+            default_invite = await self.store(
+                MediationInviteRecord.unused(provided_mediation_invitation)
+            )
+
+        return default_invite
+
+    async def mark_default_invite_as_used(self):
+        """
+        Mark the currently stored invitation as used if one exists.
+
+        :raises NoDefaultMediationInviteException:
+            if trying to mark invite as used when there is no invite stored.
+        """
+        record = await self.__retrieve()
+        if not record:
+            raise NoDefaultMediationInviteException(
+                "No default mediation invite: cannot mark it as used."
+            )
+
+        updated_record = MediationInviteRecord(record.invite, used=True)
+        await self.store(updated_record)
+
+        return updated_record
+
+    async def get_mediation_invite_record(
```

Comment:
It seems like this method should store and return the provided invitation, if any. At the moment it does not return the provided invitation if there is no stored invitation.

Reply:
Indeed, I missed a case. Good catch